.github/workflows/static.yml

```diff
@@ -38,27 +39,98 @@ jobs:
         with:
           node-version-file: '.nvmrc'
           cache: 'npm'
+      - name: Setup Pages
+        uses: actions/configure-pages@37538b280cbfab35da9dcd936a2b95f1f522d927
+      - name: Get Package version
+        id: get-package-version
+        run: |
+          echo "package_version=$(cat package.json | jq -r '.version')" >> $GITHUB_OUTPUT
+      - name: Get latest release version
+        id: get-latest-release-version
+        uses: actions/github-script@98814c53be79b1d30f795b907e553d8679345975
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          result-encoding: string
+          script: |
+            try {
+              const getReleaseResp = await github.rest.repos.getLatestRelease({
+                owner: context.repo.owner,
+                repo: context.repo.repo
+              })
+              return getReleaseResp.data.name
+            } catch(err) {
+              if (err.name == 'HttpError') {
+                console.warn("Found HttpError")
+                if (err.status == 404)  {
+                  console.error("Error 404: No previous GitHub Releases found. Defaulting to 0.0.0")
+                  return "0.0.0"
+                }
+              } else {
+                console.error(`Failed to get the latest release: ${err.message}`)
+                throw err
+              }
+            }
+      - name: Next release version
+        id: next-version
+        uses: alphagov/pay-ci/actions/next-semver@master
+        with:
+          package_version: ${{ steps.get-package-version.outputs.package_version }}
+          release_version: ${{ steps.get-latest-release-version.outputs.result }}
       - name: Install and Build
         run: |
           npm ci
           bundle exec middleman build
-      - name: Setup Pages
-        uses: actions/configure-pages@37538b280cbfab35da9dcd936a2b95f1f522d927
+          cat <<EOF > ./build/package.json
+          {
+            "name": "pay-product-page",
+            "version": "${{ steps.next-version.outputs.version }}"
+          }
+          EOF
```

Comment:
Aha - is this how the old script sneaked in a 'commit' to the package without actually committing anything?

Reply:
a small quirk of node dependencies is that they must have `package.json` even if there's nothing useful in it. Without it, when you try to `npm i` with this dependency listed you'll get an error.

The values here aren't actually important as they're never used but I kept them correct for brevity